subprojects/core/src/main/java/org/gradle/api/internal/initialization/DefaultScriptClassPathResolver.java

```diff
@@ -163,6 +154,12 @@ private static FileCollection getAnalysisResult(Configuration classpathConfigura
         }).getFiles();
     }
 
+    private static ArtifactView getOriginalDependencies(Configuration classpathConfiguration) {
+        return classpathConfiguration.getIncoming().artifactView((Action<? super ArtifactView.ViewConfiguration>) config -> {
+            config.componentFilter(it -> !isGradleApi(it));
```

Comment:
❓ We did not filter the Gradle API before, is this something we only need now?

Reply:
We don't need to pass Gradle API as input to the artifact transforms. Before we just did a bit more work unnecessary